### PR TITLE
pkg/tso: add go trace information for tso allocation

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: Close #4399

### What is changed and how does it work?

Add some trace annotation so 'go tool trace' can show the execution tracing information.

image

```commit-message
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

```
curl 'http://127.0.0.1:2379/pd/api/v1/debug/pprof/trace?seconds=3' > trace.out
go tool trace -http=:6060 trace.out
```

And observe the trace result on the web.

- No code

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: server/grpc_service.go [hunk 1]
```diff
@@ -237,7 +238,9 @@ func (s *GrpcServer) Tso(stream pdpb.PD_TsoServer) error {
 			return status.Errorf(codes.FailedPrecondition, "mismatch cluster id, need %d but got %d", s.clusterID, request.GetHeader().GetClusterId())
 		}
 		count := request.GetCount()
-		ts, err := s.tsoAllocatorManager.HandleRequest(request.GetDcLocation(), count)
+		ctx, task := trace.NewTask(ctx, "Tso")
```
Comment (lhy1024) [inline server/grpc_service.go:241]: how about "tso"？

**Review by lhy1024 — APPROVED** [review]
Thanks! but ci failed

**Comment by tiancaiamao** [comment]
Oh, I meant to provide a hotfix for a customer to debug their issues.
This one should have been made it a draft pr, it's not based on the master branch.
@lhy1024

**Comment by tiancaiamao** [comment]
Close this one in favor of https://github.com/tikv/pd/pull/6884

**Pull request closed without merging**